src/Microsoft.AspNet.Mvc.Xml/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -74,7 +66,7 @@ public XmlDictionaryReaderQuotas XmlDictionaryReaderQuotas
         }
 
         /// <inheritdoc />
-        public bool CanRead(InputFormatterContext context)
+        public override bool CanRead(InputFormatterContext context)
```

Comment:
Actually looks like we have a bug here and we don't check if we can create a serializer for the specified type.

See the WebAPI Code for this:
 public override bool CanReadType(Type type)
        {
            if (type == null)
            {
                throw Error.ArgumentNull("type");
            }

            // If there is a registered non-null serializer, we can support this type.
            // Otherwise attempt to create the default serializer.
            object serializer = GetCachedSerializer(type, throwOnError: false);

            // Null means we tested it before and know it is not supported
            return serializer != null;
        }